lms/djangoapps/learner_home/serializers.py

```diff
@@ -132,7 +132,13 @@ def get_upgradeUrl(self, instance):
         )
 
         if ecommerce_payment_page and verified_sku:
-            return f"{ecommerce_payment_page}?sku={verified_sku}"
+            query_params = {
+                'sku': verified_sku,
+                'course_run_key': str(instance.course_id)
+            }
+            encoded_params = urlencode(query_params)
+            upgrade_url = f"{ecommerce_payment_page}?{encoded_params}"
+            return upgrade_url
```

Comment:
why make this a variable? just return it like
```suggestion

             return f"{ecommerce_payment_page}?{encoded_params}"
```